### PR TITLE
Add float option to grackle

### PR DESCRIPTION
**Pull request by loikki**
Hi,

In Grackle, the code can be compiled with float or double. Therefore I have added a new variant for it.

Best regards,

### CODE REVIEW

**Comment by loikki** [comment]
I am trying to fix the coverage check, do you have a way to specify a variant during the tests?

**Comment by erimar77** [comment]
@loikki explained in the Note section here:
[http://spack.readthedocs.io/en/latest/contribution_guide.html#coverage](http://spack.readthedocs.io/en/latest/contribution_guide.html#coverage)

**Comment by loikki** [comment]
I tried to update my fork, but it is not enough. Anyway, it is not important, thank you very much

**Comment by alalazo** [comment]
No worries for coverage on package only PRs - packages are not accounted at all in coverage reports. 

The loss in coverage is an artifact, due to the fact that we test MacOS builds only once per day, and you forked from a commit that went through a nightly build. Said otherwise, the -0.55% is the coverage we lose because we didn't run unit tests on MacOS.

File: var/spack/repos/builtin/packages/grackle/package.py [hunk 1]
```diff
@@ -42,6 +42,10 @@ class Grackle(Package):
     version('2.2', 'ec49ed1db5a42db21f478285150c2ba3')
     version('2.0.1', 'a9624ad13a60c592c1a0a4ea8e1ae86d')
 
+    variant('float',
+            default=False,
+            description='Build with float')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/grackle/package.py]: Just OCD kicking in, but can you please put this variant on a single line (if it is less than 80 chars)?

Comment (loikki) [reply var/spack/repos/builtin/packages/grackle/package.py]: Done

**Comment by loikki** [comment]
Thank you for the explanation :)

**Comment by scheibelp** [comment]
Closing and reopening to restart tests

**Pull request merged**